File: src/main/java/data/structure/queue/Queue.java
package data.structure.queue;


import data.structure.queue.list.LinkedList;
import data.structure.queue.list.List;

public class Queue<E> {
	private List<E> list = new LinkedList<>();
	
	public int size() {
		return list.size();
	}

	public boolean isEmpty() {
		return list.isEmpty();
	}
	
	public void clear() {
		list.clear();
	}

	public void enQueue(E element) {
		list.add(element);
	}

	public E deQueue() {
		return list.remove(0);
	}

	public E front() {
		return list.get(0);
	}
}

File: src/main/kotlin/data/structure/queue/Queue.kt
package data.structure.queue

import data.structure.linkedlist.LinkedList
import data.structure.linkedlist.List

/**
 * 队列 ，先进行出
 */
class Queue<E> {
    private val list: List<E> = LinkedList()
    fun size(): Int {
        return list.size()
    }

    val isEmpty: Boolean
        get() = list.isEmpty

    fun clear() {
        list.clear()
    }

    fun enQueue(element: E) {
        list.add(element)
    }

    fun deQueue(): E? {
        return list.remove(0)
    }

    fun front(): E? {
        return list[0]
    }
}